src/seq.rs

```diff
@@ -225,19 +506,74 @@ fn sample_indices_cache<R>(
 #[cfg(test)]
 mod test {
     use super::*;
+    use super::IteratorExt;
+    #[cfg(feature = "alloc")]
     use {XorShiftRng, Rng, SeedableRng};
-    #[cfg(not(feature="std"))]
+    #[cfg(all(feature="alloc", not(feature="std")))]
     use alloc::Vec;
 
     #[test]
+    fn test_choose() {
+        let mut r = ::test::rng(107);
+        assert_eq!([1, 1, 1].choose(&mut r), Some(&1));
+        
+        let mut v = [2];
+        v.choose_mut(&mut r).map(|x| *x = 5);
+        assert_eq!(v[0], 5);
+
+        let v = [3, 3, 3, 3];
+        assert_eq!(v.iter().choose(&mut r), Some(&3));
+
+        let v: &[isize] = &[];
+        assert_eq!(v.choose(&mut r), None);
```

Comment:
These tests aren't very comprehensive. But I'm happy to add the tests I have in #490 over there.

Reply:
Sure, I'll look forward to it.